docs/ERROR_HANDLING.md

```diff
@@ -84,3 +84,48 @@ The following table shows the new handling logic.
 | `ibc_packet_receive`  | ⏮️ state reverted<br>✅ tx succeeds with error ack | ⏮️ state reverted<br>❌ tx fails              |
 | `ibc_packet_ack`      | ⏮️ state reverted<br>❌ tx fails                   | ⏮️ state reverted<br>❌ tx fails              |
 | `ibc_packet_timeout`  | ⏮️ state reverted<br>❌ tx fails                   | ⏮️ state reverted<br>❌ tx fails              |
+
+## Error acknowledgement formatting
+
+In case of a contract error in `ibc_packet_receive`, wasmd creates an error
+acknowledgement. The format used is a JSON object with a single top level
+`error` string such as `{"error":"some error text"}`. This format is the JSON
+serialization of the ibc-go
+[Acknowledgement](https://github.com/cosmos/ibc-go/blob/v7.0.0/proto/ibc/core/channel/v1/channel.proto#L156-L162)
+type and compatible with ICS-20.
+
+If you are using the acknowledgement types shipped with cosmwasm-std
+([#1512](https://github.com/CosmWasm/cosmwasm/issues/1512)), your protocol's
+acknowledgement is compatible with that.
```

Comment:
I don't fully understand how the `StdAck` is handled downstream. It is important to consider the state commit/rollback for "errors"  scenario [2 and 3](https://github.com/CosmWasm/wasmd/issues/697#issuecomment-992573516) 

#### Scenario 3:
If the "error" is returned to wasmd as [IBCReceiveResult.Err](https://github.com/CosmWasm/wasmvm/blob/v1.2.1/types/ibc.go#L249), then the state is rolled back and the error text [redacted (currently)](https://github.com/CosmWasm/wasmd/issues/1289) and serialized into the json format above. 
#### Scenario 2:
If the "error" is returned to wasmd as [IBCReceiveResult.Ok](https://github.com/CosmWasm/wasmvm/blob/v1.2.1/types/ibc.go#L248)[.Acknowledgement](https://github.com/CosmWasm/wasmvm/blob/v1.2.1/types/ibc.go#L260) then the data is returned as it is with the state committed. I assume `StdAck` is doing this but I may got it wrong